src/main/java/com/questhelper/QuestHelperConfig.java

```diff
@@ -38,14 +44,60 @@
 {
 	enum QuestOrdering
 	{
-		A_TO_Z,
-		OPTIMAL;
+		/** Sort quests in alphabetical order */
+		A_TO_Z(QuestOrders.sortAToZ()),
+		/** Sort quests in reverse alphabetical order */
+		Z_TO_A(QuestOrders.sortZToA()),
+		/** Sort quests according to the Optimal Quest Guide (https://oldschool.runescape.wiki/w/Optimal_quest_guide) */
+		OPTIMAL(QuestOrders.sortOptimalOrder()),
+		;
+
+		private final Comparator<QuestHelper> comparator;
+		QuestOrdering(Comparator<QuestHelper> comparator) {
+			this.comparator = comparator;
+		}
+
+		public List<QuestHelper> sort(Collection<QuestHelper> list) {
+			return list.stream().sorted(getComparator()).collect(Collectors.toList());
+		}
+
+		public Comparator<QuestHelper> getComparator() {
+			return comparator;
+		}
 	}
 
 	enum QuestFilter
 	{
-		HIDE_DONE,
-		HIDE_LACKING_REQ;
+		/** Show all quests */
+		SHOW_ALL((q,c) -> true),
+		/** Show quests where the client meets the quest requirements */
+		SHOW_MEETS_REQS((q,c) -> q.clientMeetsRequirements()),
+		/** Show all free-to-play quests */
+		FREE_TO_PLAY((q,c) -> q.getQuest().getQuestType() == Quest.Type.F2P),
+		/** Show all members' quests */
+		MEMBERS((q,c) -> q.getQuest().getQuestType() == Quest.Type.P2P),
+		/** Show all miniquets (all miniquests are members' only) */
+		MINIQUEST((q,c) -> q.getQuest().getQuestType() == Quest.Type.MINIQUEST),
+		/** Sort by difficulty */
+		BY_DIFFICULTY((q,c) -> q.getQuest().getDifficulty() == c.difficulty()),
+		/** RFD cause it ruins everything */
+		RFD((q,c) -> q.getQuest().getDifficulty() == Quest.Difficulty.RFD),
```

Comment:
I think we need it because it allows us to separate out RFD from the other quests. Due to how RFD quest state is stored as one varbit between what is basically 10 quests it's really annoying to try and separate what someone has completed from what they haven't. This way allows us to only show the 'Start - RFD' if they haven't even started RFD.
If someone has started RFD I'm not sure what it would show.

Reply:
Completed